sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/coders/AwsCoders.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.aws.coders;
+
+import com.amazonaws.ResponseMetadata;
+import com.amazonaws.http.HttpResponse;
+import com.amazonaws.http.SdkHttpMetadata;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.coders.MapCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.coders.VarIntCoder;
+import org.apache.beam.sdk.util.VarInt;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableMap;
+
+public final class AwsCoders {
+
+  private AwsCoders() {}
+
+  public static Coder<ResponseMetadata> responseMetadata() {
+    return ResponseMetadataCoder.of();
+  }
+
+  public static Coder<SdkHttpMetadata> fullSdkHttpMetadata() {
+    return FullSdkHttpMetadataCoder.of();
+  }
+
+  public static Coder<SdkHttpMetadata> minimalSdkHttpMetadata() {
+    return MinimalSdkHttpMetadataCoder.of();
+  }
+
+  private static class ResponseMetadataCoder extends Coder<ResponseMetadata>
+      implements Serializable {
+
+    private static final ResponseMetadataCoder INSTANCE = new ResponseMetadataCoder();
+
+    private ResponseMetadataCoder() {}
+
+    public static ResponseMetadataCoder of() {
+      return INSTANCE;
+    }
+
+    @Override
+    public void encode(ResponseMetadata value, OutputStream outStream)
+        throws CoderException, IOException {
+      MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())
```

Comment:
The encoding of this changed where before we assumed this would always exist but now we handle the null case but we don't actually return the same object that was provided when decoding. Would you rather handle the null case by wrapping the response metadata coder returned in `responseMetadata` with the [NullableCoder](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/NullableCoder.java)?